PayPalDataCollector/src/main/java/com/braintreepayments/api/PayPalDataCollector.java

```diff
@@ -0,0 +1,131 @@
+        this.magnesInternalClient = magnesInternalClient;
+        this.uuidHelper = uuidHelper;
+    }
+
+    String getPayPalInstallationGUID(Context context) {
+        return uuidHelper.getInstallationGUID(context);
+    }
+
+    /**
+     * Gets a Client Metadata ID at the time of payment activity. Once a user initiates a PayPal payment
+     * from their device, PayPal uses the Client Metadata ID to verify that the payment is
+     * originating from a valid, user-consented device and application. This helps reduce fraud and
+     * decrease declines. This method MUST be called prior to initiating a pre-consented payment (a
+     * "future payment") from a mobile device. Pass the result to your server, to include in the
+     * payment request sent to PayPal. Do not otherwise cache or store this value.
+     *
+     * @param context          Android Context
+     * @param configuration    the merchant configuration
+     */
+    @MainThread
+    String getClientMetadataId(Context context, Configuration configuration) {
+        PayPalDataCollectorRequest request = new PayPalDataCollectorRequest()
+                .setApplicationGuid(getPayPalInstallationGUID(context));
+
+        return getClientMetadataId(context, request, configuration);
+    }
+
+    /**
+     * Gets a Client Metadata ID at the time of payment activity. Once a user initiates a PayPal payment
+     * from their device, PayPal uses the Client Metadata ID to verify that the payment is
+     * originating from a valid, user-consented device and application. This helps reduce fraud and
+     * decrease declines. This method MUST be called prior to initiating a pre-consented payment (a
+     * "future payment") from a mobile device. Pass the result to your server, to include in the
+     * payment request sent to PayPal. Do not otherwise cache or store this value.
+     *
+     * @param context       Android Context.
+     * @param request       configures what data to collect.
+     * @param configuration the merchant configuration
+     */
+    @MainThread
+    String getClientMetadataId(Context context, PayPalDataCollectorRequest request, Configuration configuration) {
+        return magnesInternalClient.getClientMetadataId(context, configuration, request);
+    }
+
+    /**
+     * Collects device data based on your merchant configuration.
+     * <p>
+     * We recommend that you call this method as early as possible, e.g. at app launch. If that's too early,
+     * call it at the beginning of customer checkout.
+     * <p>
+     * Use the return value on your server, e.g. with `Transaction.sale`.
+     *
+     * @param context  Android Context
+     * @param callback {@link PayPalDataCollectorCallback}
+     */
+    public void collectDeviceData(@NonNull final Context context, @NonNull final PayPalDataCollectorCallback callback) {
+        collectDeviceData(context, null, callback);
+    }
+
+    /**
+     * Collects device data based on your merchant configuration.
```

Comment:
Clarification here is that collection isn't based on merchant config.

Maybe `Collects device data for PayPal APIs`